sopel/modules/reddit.py

```diff
@@ -236,9 +236,58 @@ def comment_info(bot, trigger, match):
     bot.say(message)
 
 
+def subreddit_sorting(bot, trigger, s, sorting):
+    if sorting == 'new':
+        submissions = list(s.new())
+    elif sorting == 'top':
+        submissions = list(s.top())
+    elif sorting == 'hot':
+        submissions = list(s.hot())
+    elif sorting == 'controversial':
+        submissions = list(s.controversial())
+    elif sorting == 'gilded':
+        submissions = list(s.gilded())
+    elif sorting == 'rising':
+        submissions = list(s.rising())
+    elif sorting == 'sticky':
+        try:
+            submissions = [s.sticky()]
+        except prawcore.exceptions.NotFound:
+            bot.say("r/" + s.display_name + " appears to not have a stickied post!")
+            return
+    elif sorting == 'random':
+        submissions = [s.random()] or []
+    else:
+        return
+
+    if not len(submissions):
+        bot.say("r/" + s.display_name + ' ' + sorting + " appears to have no items!")
+        return NOLIMIT
+
+    if sorting != 'sticky':
+        submissions_filter = []
+        for submission in submissions:
+            if not submission.stickied:
+                submissions_filter.append(submission)
+        submissions = submissions_filter
+    submission = submissions[0]
+
+    link = "https://reddit.com/r/" + s.display_name + "/comments/" + str(submission)
```

Comment:
```suggestion
    link = "https://reddit.com/r/" + s.display_name + "/comments/" + submission.id
```

It strikes me as fragile to rely on the human-readable string version when we already have the ID as an attribute.

I'd probably use `.format()` here instead of concatenation with `+` but… this isn't complicated enough to really care.